clients/da-vinci-client/src/main/java/com/linkedin/davinci/client/DaVinciRecordTransformerUtility.java

```diff
@@ -127,35 +133,11 @@ public final void onRecovery(
           } catch (IOException e) {
             throw new RuntimeException(e);
           }
-          return getOutputValueDeserializer().deserialize(decompressedValueBytes);
+          return outputValueDeserializer.deserialize(decompressedValueBytes);
         });
 
         recordTransformer.processPut(lazyKey, lazyValue);
       }
     }
   }
-
-  public AvroGenericDeserializer<K> getKeyDeserializer() {
-    if (keyDeserializer == null) {
-      Schema keySchema = recordTransformer.getKeySchema();
-      keyDeserializer = new AvroGenericDeserializer<>(keySchema, keySchema);
-    }
-    return keyDeserializer;
-  }
-
-  public AvroGenericDeserializer<O> getOutputValueDeserializer() {
-    if (outputValueDeserializer == null) {
-      Schema outputValueSchema = recordTransformer.getOutputValueSchema();
-      outputValueDeserializer = new AvroGenericDeserializer<>(outputValueSchema, outputValueSchema);
-    }
-    return outputValueDeserializer;
-  }
-
-  public AvroSerializer<O> getOutputValueSerializer() {
-    if (outputValueSerializer == null) {
-      Schema outputValueSchema = recordTransformer.getOutputValueSchema();
-      outputValueSerializer = new AvroSerializer<>(outputValueSchema);
-    }
-    return outputValueSerializer;
-  }
```

Comment:
Nice. This lazy init code was not threadsafe anyway, so one less thing to go wrong...